subprojects/base-services/src/main/java/org/gradle/api/JavaVersion.java

```diff
@@ -54,7 +54,7 @@ public static JavaVersion toVersion(Object value) throws IllegalArgumentExceptio
         }
 
         String name = value.toString();
-        Matcher matcher = Pattern.compile("(\\d{1,2})(-.+)?").matcher(name);
+        Matcher matcher = Pattern.compile("(\\d{1,2})([-.].+)?").matcher(name);
```

Comment:
Instead of using `[-.]` we should probably be using `\D` as we do for the version starting with `1.` below. We should also move the `1.` version check before the "Any number." check, so the old versions are still detected correctly I guess.

Reply:
Yes, you're right. This fix is too haste. Let me polish it.